lib/core.ps1

```diff
@@ -839,18 +839,28 @@ function is_scoop_outdated() {
 }
 
 function substitute($entity, [Hashtable] $params, [Bool]$regexEscape = $false) {
-    if ($entity -is [Array]) {
-        return $entity | ForEach-Object { substitute $_ $params $regexEscape}
-    } elseif ($entity -is [String]) {
-        $params.GetEnumerator() | ForEach-Object {
-            if($regexEscape -eq $false -or $null -eq $_.Value) {
-                $entity = $entity.Replace($_.Name, $_.Value)
-            } else {
-                $entity = $entity.Replace($_.Name, [Regex]::Escape($_.Value))
+    switch ($entity.GetType().Name) {
+        'String' {
+            $params.GetEnumerator() | ForEach-Object {
+                if ($regexEscape -eq $false -or $null -eq $_.Value) {
+                    $entity = $entity.Replace($_.Name, $_.Value)
+                } else {
+                    $entity = $entity.Replace($_.Name, [Regex]::Escape($_.Value))
+                }
             }
+            break
```

Comment:
Are breaks needed? As soon as variable is set switch will be ended naturally (in all branches)

Reply:
Break make sense only when you need to conditionally break from whole switch execution.

Reply:
No... `switch` will match all posible variables and run from top to bottom... And I've bugfixed a long time when I first used `switch` in PowerShell... So best solution is `break`.

Reply:
Yeah it is. But there should not be multiple matches right? 

As if you would have something like:

```
switch -wildcard ('testing') {
     'test*' { Write-Host 'A' }
      '*ing' { Write-Host 'B' }
}
```
This would print `A` and `B` as it will fit both, but in this case there should not be any entity, which is String  and Object[] (or PSCustomObject)

Reply:
Latest `break` could be omitted, but for better readability, I keep it.

Reply:
Ok. Keep it. I was just curious if there is some potentional use case for it.

Reply:
Got it, `xxx.GetType().Name` is just string, so multi match is imposible. I met error when I match between Type, and there are some inheritance.